.github/workflows/cancel_workflows_for_pr.yaml

```diff
@@ -12,22 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-name: Cancel workflow on PR
+name: Cancel workflows on failing CI
 on:
     workflow_dispatch:
-      inputs:
-        pull_request_id:
-          description: 'PR number to consider'
-          required: true
-          type: number
-        commit_sha:
-          description: 'Cancel runs for this specific SHA'
-          required: true
-          type: string
+    schedule:
+        - cron: "*/10 * * * *"
```

Comment:
possibly a dumb question, but this cron job would be running based on what is in master and not a possible uploaded PR for review.

I don't think so, but also want to double check that that someone couldn't maliciously start cancelling jobs by uploading a PR that contains a change to `scripts/tools/cancel_workflows_for_pr.py`

Reply:
This will only run on master CI, so changing the python script would only take effect if the PR is approved and merged.